dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/CompanyOwnershipChecker.kt

```diff
@@ -56,4 +58,44 @@ class CompanyOwnershipChecker(
         val companyId = dataMetaInformationManager.getDataMetaInformationByDataId(dataId).company.companyId
         return isCurrentUserCompanyOwnerForCompany(companyId)
     }
+
+    /**
+     * Method to check whether a given company has at least one company owner
+     * @param companyId the ID of the company
+     * @return a Boolean indicating whether the company has at least one company owner
+     */
+    @Transactional(readOnly = true)
+    fun companyExistsAndHasNoOwner(companyId: String): Boolean {
```

Comment:
Ich schlage vor die Funktion `hasCompanyNoOwner` zu nennen.

Außerdem checkst du glaube ich nicht, ob die company existiert.
Wenn die company nicht existiert, wird es glaube ich einen internal server error geben.

Reply:
here the function return true when the assignment chekcer gives back []. This makes sure both conditions are ture. if the company didn't exits, we wouldn't have this empty list.

Reply:
let me investigate this :)

Reply:
Ich denke so wie es aktuell ist, ist es noch nicht ganz richtig.

Mein Vorschlag wäre das hier:

![image](https://github.com/d-fine/Dataland/assets/99255296/f329bebf-0db2-4609-a862-22885c8c9d5b)

Reply:
applied

Reply:
Finde den Funktionsnamen `doesCompanyExistsAndHaveNoOwner` etwas schwierig.

Was hälst du von `isCompanyExistantAndWithoutOwner" ?

Reply:
done